apps/science/main.cpp

```diff
@@ -1,12 +1,155 @@
+Motor indexerMotor(MTR_PWM_1, MTR_DIR_1, false);
+Motor elevatorMotor(MTR_PWM_2, MTR_DIR_2, false);
+
+// Servo
+Servo coverServo(SRVO_PWM_1, ScienceConfig::coverServoType, ScienceConfig::coverServoRange,
+                 ScienceConfig::coverServoMaxPulse, ScienceConfig::coverServoMinPulse);
+Servo diggerServo(SRVO_PWM_2, ScienceConfig::diggerServoType, ScienceConfig::diggerServoRange,
+                  ScienceConfig::diggerServoMaxPulse, ScienceConfig::diggerServoMinPulse);
+
+// encoders
+EncoderAbsolute_PWM elevatorEncoder(ScienceConfig::elevatorEncoderConfig);
+EncoderRelative_Quadrature indexerEncoder(ScienceConfig::centrifugeEncoderConfig);
+// limit switches
+
+DigitalIn indexerLimFront(LIM_SW_1);
+DigitalIn indexerLimBack(LIM_SW_2);
+DigitalIn elevatorLimTop(LIM_SW_3);
+DigitalIn elevatorLimBottom(LIM_SW_4);
+
+// Actuators
+ActuatorController indexerActuator(ScienceConfig::indexerActuatorConfig, indexerMotor, indexerEncoder, indexerLimFront,
+                                   indexerLimBack);
+ActuatorController elevatorActuator(ScienceConfig::diggerLiftActuatorConfig, elevatorMotor, elevatorEncoder,
+                                    elevatorLimBottom, elevatorLimTop);
+
+// I2C
+
+MoistureSensor moistureSensor = MoistureSensor(TEMP_MOIST_I2C_SDA, TEMP_MOIST_I2C_SCL);
 
 DigitalOut led1(LED1);
+DigitalOut ledR(LED_R);
+DigitalOut ledG(LED_G);
+DigitalOut ledB(LED_B);
+
+CAN can(CAN1_RX, CAN1_TX, ROVER_CANBUS_FREQUENCY);
+
+static mbed_error_status_t setMotionData(CANMsg &msg) {
+  float motionData;
+  msg.getPayload(motionData);
 
+  switch (msg.id) {
+    case CANID::SET_INDEXER_POS:
+      return indexerActuator.setMotionData(motionData);
+    case CANID::SET_ELEVATOR_POS:
+      return elevatorActuator.setMotionData(motionData);
+    case CANID::SET_COVER_POS:
+      return coverServo.set_position(motionData);
+    case CANID::SET_DIGGER_POS:
+      return diggerServo.set_position(motionData);
+    case CANID::SET_MOISTURE_SENSOR:
+      return moistureSensor.Is_Initialized();
+    default:
+      return MBED_ERROR_INVALID_ARGUMENT;
+  }
+}
 const int k_interval_ms = 500;
 
+// CAN Threads
+Thread rxCANProcessorThread;
+Thread txCANProcessorThread;
+
+static CANMsg::CANMsgHandlerMap canHandleMap = {{CANID::SET_INDEXER_POS, setMotionData},
+                                                {CANID::SET_ELEVATOR_POS, setMotionData},
+                                                {CANID::SET_COVER_POS, setMotionData},
+                                                {CANID::SET_DIGGER_POS, setMotionData},
+                                                {CANID::SET_MOISTURE_SENSOR, setMotionData}};
+
+void rxCANProcessor() {
+  CANMsg rxMsg;
+
+  while (true) {
+    if (can.read(rxMsg)) {
+      if (canHandleMap.count(rxMsg.id) > 0) {
+        canHandleMap[rxMsg.id](rxMsg);
+      } else {
+        // ruh roh
+      }
+    }
+  }
+}
+
+// Send outgoing CAN messages
+void txCANProcessor() {
+  const int txPeriod_millisec = 500;
```

Comment:
try to have const variables as all caps and you can also move them out of the functions incase they are needed elsewhere